sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVFileFormat.scala

```diff
@@ -145,6 +145,8 @@ class CSVFileFormat extends TextBasedFileFormat with DataSourceRegister {
   override def equals(other: Any): Boolean = other.isInstanceOf[CSVFileFormat]
 
   override def supportDataType(dataType: DataType): Boolean = dataType match {
+    case _: BinaryType => false
```

Comment:
Did it work before? Disabling it makes sense at a glance but would be great if we can elabourate the reason behind so make sure it doesn't mistakenly break a reasonable workload that worked before.

Reply:
As description in [SPARK-42237](https://issues.apache.org/jira/browse/SPARK-42237): 
- when writing, it works, but content written in csv files is not the actual value of the binary column.
- when reading, it doesn't work, throws unsupported exception in UnivocityParser.

It causes that if users write csv tables with binary columns, like using udf UnBase64, these tables can't be read follow-up.